spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyAuthTokenVerifier.java

```diff
@@ -20,166 +20,124 @@
+            log.debug("Verifying the authentication {}.", authentication.getName());
+            return createApiKeyAuthToken(authentication);
+        } catch (Exception e) {
+            log.info("During the authentication process this exception occurred: ", e);
         }
+        throw new SessionAuthenticationException("The authentication process failed.");
     }
 
-    private boolean isApiKeyValid(String apiKeyValue) throws AuthenticationException {
-
-        try {
-            log.debug(String.format("checking APIKEY value '%s' with CoatRack admin server at '%s'", apiKeyValue, adminBaseUrl));
-
-            String urlToSearchForApiKeys = securityUtil.attachGatewayApiKeyToUrl(
-                    adminBaseUrl + adminResourceToSearchForApiKeys + apiKeyValue);
-
-            ResponseEntity<ApiKey> resultOfApiKeySearch = restTemplate.getForEntity(urlToSearchForApiKeys, ApiKey.class);
-
-            if (resultOfApiKeySearch != null) {
-                ApiKey apiKey = resultOfApiKeySearch.getBody();
-                if (apiKey != null) {
-                    log.debug("API key was found by CoatRack admin: " + apiKey);
-
-                    if (Date.valueOf(LocalDate.now()).after(apiKey.getValidUntil())) {
-                        throw new CredentialsExpiredException("Api key is expired");
-                    }
-                    if (apiKey.getDeletedWhen() != null) {
-                        return false;
-                    }
-                } else {
-                    log.debug("API key value could not be found by CoatRack Admin: " + apiKeyValue);
-                    return false;
-                }
-                return true;
-            } else {
-                log.error("Communication with Admin server failed, result is: " + resultOfApiKeySearch);
-                throw new AuthenticationServiceException("Communication with auth server failed");
-            }
-        } catch (HttpClientErrorException e) {
-            if (e.getStatusCode().equals(HttpStatus.NOT_FOUND)) {
-                log.debug("API key value is invalid: " + apiKeyValue);
-                return false;
-            } else {
-                log.error("Error when communicating with auth server", e);
-                throw new AuthenticationServiceException("Communication with auth server failed");
-            }
-        }
+    private Authentication createApiKeyAuthToken(Authentication authentication) {
+        String apiKeyValue = getApiKeyValue(authentication);
+        //TODO this is just a workaround for now: check for fixed API key to allow CoatRack admin access
+        if (isAdminsKey(apiKeyValue)) {
+            return createAdminsAuthToken(apiKeyValue);
+        } else
+            return verifyApiKeyAndIfAuthorizedCreateConsumerAuthToken(apiKeyValue);
     }
 
-    private ServiceApi getServiceApiByApiKey(String apiKeyValue) {
+    private String getApiKeyValue(Authentication authentication) {
```

Comment:
Maybe a better name would be `extractApiKeyValueFromAuthentication`

Reply:
This function does not exist anymore. Instead it was taken advantage of the fact that an API keeps its service as a field value. So it is only necessary to search for the API key.

Reply:
Done.